js/derivadex.js

```diff
@@ -0,0 +1,599 @@
+                'base': base,
+                'quote': 'USD',
+                'settle': 'USDC',
+                'baseId': base.toLowerCase (),
+                'quoteId': 'usd',
+                'settleId': 'usdc',
+                'type': 'swap',
+                'spot': 'false',
+                'margin': 'false',
+                'swap': 'true',
+                'future': 'false',
+                'option': 'swap',
+                'active': 'true',
+                'contract': 'true',
+                'linear': 'true',
+                'inverse': 'false',
+                'taker': '0.002',
+                'maker': '0.000',
+                'precision': {
+                    'amount': 6,
+                    'price': 6,
+                    'quote': 6,
+                },
+                'limits': {
+                    'leverage': {
+                        'min': 1,
+                        'max': 3,
+                    },
+                    'amount': {
+                        'min': market['value']['tickSize'],
+                        'max': undefined,
+                    },
+                    'price': {
+                        'min': market['value']['tickSize'],
+                        'max': undefined,
+                    },
+                    'cost': {
+                        'min': market['value']['minOrderSize'],
+                        'max': market['value']['maxOrderNotional'],
+                    },
+                },
+                'info': market,
+            };
+        });
+    }
+
+    async fetchCurrencies (params = {}) {
+        /**
+         * @method
+         * @name derivadex#fetchCurrencies
+         * @description fetches all available currencies on an exchange
+         * @param {object} params extra parameters specific to the derivadex api endpoint
+         * @returns {object} an associative dictionary of currencies
+         */
+        const networks = {};
+        networks['ERC20'] = {
+            'info': undefined,
+            'id': 'ETH',
+            'network': this.networkIdToCode ('ETH'),
+            'active': true,
+            'deposit': true,
+            'withdraw': true,
+            'fee': undefined,
+        };
+        return [
+            {
+                'id': 'usdc',
+                'code': 'USDC',
+                'name': 'USDC',
+                'active': true,
+                'fee': 0,
+                'precision': 2,
+                'deposit': true,
+                'withdraw': true,
+                'limits': {
+                    'deposit': {
+                        'min': 1000,
+                        'max': 1000000,
+                    },
+                },
+                'networks': networks,
+                'info': undefined,
+            },
+            {
+                'id': 'ddx',
+                'code': 'DDX',
+                'name': 'DDX',
+                'active': false,
+                'fee': 0,
+                'precision': 2,
```

Comment:
what does precision mean in this context? if there are fields you are unsure about or are putting placeholders for now, it would be good to annotate them so that I can tell which are question marks vs. which you're pretty sure are mean to be there